pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -278,6 +278,12 @@ func (r *ReconcileNetworkAddonsConfig) renderObjects(networkAddonsConfig *opv1al
 		}
 	}
 
+	// Clean Up any outdated obsoleted objects
```

Comment:
Please change the comment:

```
// Perform any special object changes that are impossible to do with regular Apply. e.g. Remove outdated objects and objects that cannot be modified by Apply method due to incompatible changes. 
```

Move this after the `Render` execution (in case Render fails, we should not do any of this).

Reply:
I like this comment rephrase.